app/assets/javascripts/components/general_button.es6.jsx

```diff
@@ -0,0 +1,58 @@
+class GeneralButton extends React.Component {
+
+  static get propTypes() {
+    return {
+      action: React.PropTypes.func,
+      route: React.PropTypes.string,
+      content: React.PropTypes.string.isRequired,
+    };
+  }
+
+  static get defaultProps() {
+    return {
+      action: null,
+      route: '',
+      content: '',
+    };
+  }
+
+  get styles() {
+    return {
+      container: {
+        display: 'inline-block',
+        verticalAlgin: 'middle',
+        position: 'relative',
+        padding: '6px 12px',
+        color: 'white',
+        backgroundColor: '#68B1DE',
+        borderBottom: '2px solid',
+        borderColor: '#28719E',
+        borderRadius: 2,
+        fontSize: '14px',
+      },
+    };
+  }
+
+  handleClick(event) {
+    if (this.props.route === '') {
+      event.preventDefault();
+      event.stopPropagation();
+      if (this.prop.action !== null) {
+        this.props.action();
+      }
+    }
+  }
+
+  render () {
+    console.log(this.props.route);
+    return (
+      <a
+        href={this.props.route}
+        onClick={this.handleClick.bind(this)}
```

Comment:
Great question (I was hoping someone would catch this!). Essentially, in javascript es6 you'll lost your scope when you attach a callback. This means that when `handleClick` gets invoked, `this` won't reference the component instance - it'll reference the `window` of the browser. To avoid that and have access to the component instance with `this`, we use `.bind(this)` to maintain scope. Later on we might use something called the rocket as another way to do this.

If none of that made sense, just note that there's a reason for the `.bind(this)` when you attach a callback listener and I'll go more in depth next meeting!